packages/bundler-webpack/src/config/manifestPlugin.ts

```diff
@@ -20,5 +22,53 @@ export async function addManifestPlugin(opts: IOpts) {
         ...userConfig.manifest,
       },
     ]);
+    if (userConfig.ssr && opts.webpackManifest) {
+      config
+        .plugin('manifest-copy-plugin')
+        .use(WebpackAssetsManifestCopyPlugin, [
+          opts.userConfig,
+          opts.webpackManifest,
+        ]);
+    }
+  }
+}
+
+// This webpack plugin can copy the assets-manifest.json into the webpackManifest variable.
+// The webpackManifest variable will be passed to esbuild for building umi.server.js
+class WebpackAssetsManifestCopyPlugin {
+  userConfig: any;
+  webpackManifest: Map<string, string>;
+
+  constructor(userConfig: any, webpackManifest: Map<string, string>) {
+    this.userConfig = userConfig;
+    this.webpackManifest = webpackManifest;
+  }
+
+  apply(compiler: webpack.Compiler) {
+    compiler.hooks.compilation.tap(
+      'WebpackAssetsInjectPlugin',
+      (compilation) => {
+        compilation.hooks.afterProcessAssets.tap(
+          'WebpackAssetsInjectPlugin',
+          (a) => {
+            const manifestKey = Object.keys(a).find(
+              (key) =>
+                key === this.userConfig.manifest.fileName ||
+                key === 'asset-manifest.json',
```

Comment:
```suggestion
              (key) => ['asset-manifest.json', this.userConfig.manifest.fileName].includes(key)
```

`asset-manifest.json` 这个变量不抽象么，源码里有好几处都用了